php/api/endpoints/class-coauthors-controller.php

```diff
@@ -0,0 +1,401 @@
+		$this->coauthors_plus = $coauthors_plus;
+	}
+
+	/**
+	 * Register Rest Routes
+	 *
+	 * @since 3.6.0
+	 */
+	public function register_routes(): void {
+		$this->register_coauthors_route();
+		$this->register_coauthor_route();
+	}
+
+	/**
+	 * Register Co-Authors Route
+	 *
+	 * Provide a post ID as an integer to retrieve an array of associated co-authors.
+	 *
+	 * Example: `/wp-json/coauthors/v1/coauthors?post_id=11111`
+	 *
+	 * @since 3.6.0
+	 */
+	public function register_coauthors_route(): void {
+		register_rest_route(
+			'coauthors/v1',
+			'/coauthors',
+			array(
+				'args' => array(
+					'post_id' => array(
+						'description'       => __( 'Unique identifier for a post.', 'co-authors-plus' ),
+						'type'              => 'integer',
+						'required'          => true,
+						'validate_callback' => function( $post_id ): bool {
+							return 0 !== absint( $post_id );
+						},
+						'sanitize_callback' => function( $post_id ): int {
+							return absint( $post_id );
+						},
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permission_check' ),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Register Co-Author Route
+	 *
+	 * Provide a user nicename as a hyphen-separated string to retrieve a single co-author.
+	 *
+	 * Example: `/wp-json/coauthors/v1/coauthors/user-nicename`
+	 *
+	 * @since 3.6.0
+	 */
+	public function register_coauthor_route(): void {
+		register_rest_route(
+			'coauthors/v1',
+			'/coauthors/(?P<user_nicename>[\w-]+)',
+			array(
+				'args' => array(
+					'user_nicename' => array(
+						'description'       => __( 'Nicename / slug for co-author.', 'co-authors-plus' ),
+						'type'              => 'string',
+						'required'          => true,
+						'validate_callback' => function( $slug ): bool {
+							return is_string( $slug );
+						},
+						'sanitize_callback' => function( $slug ) {
+							return sanitize_title( $slug );
+						},
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => '__return_true',
+				),
+			)
+		);
+	}
+
+	/**
+	 * Get Item
+	 *
+	 * @since 3.6.0
+	 * @param WP_REST_Request $request
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function get_item( $request ) {
+
+		$coauthor = $this->coauthors_plus->get_coauthor_by(
+			'user_nicename',
+			$request->get_param( 'user_nicename' )
+		);
+
+		if ( ! is_object( $coauthor ) ) {
+			return new WP_Error(
+				'rest_not_found',
+				__( 'Sorry, we could not find that co-author.', 'co-authors-plus' ),
+				array( 'status' => 404 )
+			);
+		}
+
+		if ( ! self::is_coauthor( $coauthor ) ) {
+			return new WP_Error(
+				'rest_unusable_data',
+				__( 'Sorry, an unusable response was produced.', 'co-authors-plus' ),
+				array( 'status' => 406 )
```

Comment:
@douglas-johnson Any particular reason for a `406` here? It seems to be defined as [related to `Accept` headers](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#406) which I'm not sure apply here. Can this be a `404` as well?

Reply:
Sorry about that, addressed in https://github.com/Automattic/Co-Authors-Plus/pull/997/commits/062a2d6b15fe610d90f34085a3d7f14cf73765c6